test/apps/iodemo/io_demo.cc

```diff
@@ -650,19 +651,33 @@ class P2pDemoCommon : public UcxContext {
             }
         }
 
-        void init(size_t data_size, void *ext_buf)
+        void init(size_t data_size, BufferMemoryPool<Buffer> &chunk_pool,
+                  void *ext_buf, bool validate)
         {
             assert(ext_buf != NULL);
             assert(_iov.empty());
             assert(_extra_buf == NULL);
+            assert(_bogus_buf == NULL);
 
             _data_size = data_size;
             _extra_buf = ext_buf;
+            _bogus_buf = new Buffer(NULL, 0, chunk_pool, UCS_MEMORY_TYPE_HOST,
+                                    NULL, NULL);
+            if (_bogus_buf == NULL) {
+                throw std::bad_alloc();
+            }
+            _validate  = validate;
         }
 
         inline Buffer &operator[](size_t i) const
         {
-            return *_iov[i];
+            if (!_iov.empty()) {
+                return *_iov[i];
+            } else {
+                assert(_extra_buf != NULL);
+                assert(_bogus_buf != NULL);
+                return *_bogus_buf;
```

Comment:
can we make it a static local / member variable to avoid allocating it every time?

Reply:
done